TimeThrottleIntegrationTests testTimeThrottleDefaults failure

### CI Link

https://gradle-enterprise.elastic.co/s/5myw2twx435xi

### Repro line

`./gradlew ':x-pack:plugin:watcher:internalClusterTest' --tests "org.elasticsearch.xpack.watcher.actions.TimeThrottleIntegrationTests.testTimeThrottleDefaults" -Dtests.seed=EC68403920230004 -Dtests.locale=es -Dtests.timezone=Asia/Oral -Druntime.java=20`

### Does it reproduce?

No

### Applicable branches

main

### Failure history

In the past 7 days this is the first time it's failing `https://gradle-enterprise.elastic.co/scans/tests?search.timeZoneId=Europe/Bucharest&tests.container=org.elasticsearch.xpack.watcher.actions.TimeThrottleIntegrationTests&tests.test=testTimeThrottleDefaults`

### Failure excerpt

```
org.elasticsearch.xpack.watcher.actions.TimeThrottleIntegrationTests > testTimeThrottleDefaults FAILED
    java.lang.AssertionError: 
    Expected: is one of {<3L>, <4L>}
         but: was <2L>
        at __randomizedtesting.SeedInfo.seed([EC68403920230004:504646CBD1BA7D6E]:0)
        at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
        at org.junit.Assert.assertThat(Assert.java:956)
        at org.junit.Assert.assertThat(Assert.java:923)
        at org.elasticsearch.xpack.watcher.actions.TimeThrottleIntegrationTests.assertTotalHistoryEntries(TimeThrottleIntegrationTests.java:120)
        at org.elasticsearch.xpack.watcher.actions.TimeThrottleIntegrationTests.testTimeThrottleDefaults(TimeThrottleIntegrationTests.java:84)
```

--- a/x-pack/plugin/watcher/src/internalClusterTest/java/org/elasticsearch/xpack/watcher/actions/TimeThrottleIntegrationTests.java
+++ b/x-pack/plugin/watcher/src/internalClusterTest/java/org/elasticsearch/xpack/watcher/actions/TimeThrottleIntegrationTests.java
@@ -111,12 +111,17 @@ public class TimeThrottleIntegrationTests extends AbstractWatcherIntegrationTest
         });
     }
 
-    private void assertTotalHistoryEntries(String id, long expectedCount) {
-        SearchResponse searchResponse = client().prepareSearch(HistoryStoreField.DATA_STREAM + "*")
-            .setSize(0)
-            .setSource(new SearchSourceBuilder().query(QueryBuilders.boolQuery().must(termQuery("watch_id", id))))
-            .get();
+    private void assertTotalHistoryEntries(String id, long expectedCount) throws Exception {
+        assertBusy(() -> {
+            // Watcher history is now written asynchronously, so we check this in an assertBusy
+            ensureGreen(HistoryStoreField.DATA_STREAM);
+            SearchResponse searchResponse = client().prepareSearch(HistoryStoreField.DATA_STREAM + "*")
+                .setSize(0)
+                .setSource(new SearchSourceBuilder().query(QueryBuilders.boolQuery().must(termQuery("watch_id", id))))
+                .get();
 
-        assertThat(searchResponse.getHits().getTotalHits().value, is(oneOf(expectedCount, expectedCount + 1)));
+            assertThat(searchResponse.getHits().getTotalHits().value, is(oneOf(expectedCount, expectedCount + 1)));
+        });
+
     }
 }

